packages/oats-runtime/src/server.ts

```diff
@@ -131,11 +131,8 @@ function voidify(value: object | undefined | null) {
 function cleanHeaders<H>(mode: Mode, maker: Maker<any, H>, headers: object) {
   // note: we now expect that on client side headers are type conforming so do not need to lowercase those
   // this is slightly breaking change is somebody has subverted the type checker
-  const normalized = voidify(mode === 'server' ? lowercaseObject(headers) : headers);
-  const acceptsNull = maker(null);
-  if (acceptsNull.isSuccess()) {
-    return acceptsNull.success();
-  }
+  const normalized = mode === 'server' ? lowercaseObject(headers) : headers;
```

Comment:
I can still keep the old before (acceptsNull check which i removed). Just removing voidify() will fix the bug with the empty headers.
One thing i don't like about the current implementation is that headers value is typed as `void`, but its value is `null` (this is in case no header parameters is specified in the spec). I'm not a TS expert, but to my knowledge undefined can be used as void type, not null.

Reply:
Btw same is true for `ctx.params` - `null` value, `void` type.